test/core/plugin_test.py

```diff
@@ -1,29 +1,61 @@
 """Plugin related tests."""
+import logging
 import pytest
+import sys
 
-from sqlfluff.core.plugin.host import get_plugin_manager
+from sqlfluff.core.plugin.host import get_plugin_manager, purge_plugin_manager
 from sqlfluff.core.config import FluffConfig
+from sqlfluff.utils.testing.logging import fluff_log_catcher
 
 
 def test__plugin_manager_registers_example_plugin():
-    """Test that the example plugin is registered."""
-    plugin_manager = get_plugin_manager()
-    # The plugin import order is non-deterministic.
-    # Use sets in case the dbt plugin (or other plugins) are
-    # already installed too.
-    installed_plugins = set(
-        plugin_module.__name__ for plugin_module in plugin_manager.get_plugins()
-    )
+    """Test that the example plugin is registered.
+
+    This test also tests that warnings are raised on the import of
+    plugins which have their imports in the wrong place (e.g. the
+    example plugin). That means we need to make sure the plugin is
+    definitely reimported at the start of this test, so we can see
+    any warnings raised on imports.
+
+    To do this we clear the plugin manager cache and also forcibly
+    unload the example plugin modules if they are already loaded.
+    This ensures that we can capture any warnings raised by importing the
+    module.
+    """
+    purge_plugin_manager()
+    try:
+        del sys.modules["sqlfluff_plugin_example"]
+    except KeyError:
```

Comment:
why are we doing try, excepts in a test?

Reply:
Ah - good question. It's because we may or may not be able to _remove_ the module depending on whether it's already been imported by _a different test_. The imported modules are shared between tests.

I'll add a comment to that effect.